### PR TITLE
chore: Update readme with standard format and examples

### LINKED ISSUE #100: Add comprehensive and consistent readme
Update readme to be consistent with the template here: https://github.com/open-feature/.github/tree/main/templates/READMEs

Definition of done:

- README structure matches template
- Examples are up-to-date and added for all `TODOs` in template
- details only pertaining to developers of the SDK are moved to CONTRIBUTING.md

See similar Java PR: https://github.com/open-feature/java-sdk/pull/382

### PR DESCRIPTION
**Pull request by jamescarr**
## This PR

- Uses the standard readme template
- added additional examples for context

### Related Issues

Fixes #100

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -71,22 +64,111 @@ open_feature_api.set_provider(NoOpProvider())
...
+
+# merge second context
+client = get_client(name="No-op Provider", version="0.5.2")
+client.get_string_value("email", None, request_context)
+
+```
+### Events
+
+Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
+
```
Comment (beeme1mr) [inline readme.md:122]: I would recommend either adding a section saying that `Events` haven't been implemented and possibly adding a link to a GitHub issue or removing this section for now.

File: readme.md
Comment (beeme1mr) [inline readme.md:1]: Could you update the title based on the template as well?

File: readme.md (hunk 1 above)
Comment (jamescarr) [reply readme.md]: I added an events issue and linked it, as well as linking the hooks GitHub issue as those are currently unimplemented as well.

File: readme.md
Comment (jamescarr) [reply readme.md:1]: Yep! Just added the banner and title format. Didn't notice it!

File: readme.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+<p align="center">
+  <picture>
+    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/white/openfeature-horizontal-white.svg">
+    <source media="(prefers-color-scheme: light)" srcset="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/black/openfeature-horizontal-black.svg">
+    <img align="center" alt="OpenFeature Logo">
+  </picture>
+</p>
+
+<!-- TODO: update title -->
+<h2 align="center">OpenFeature python-sdk</h2>
```
Comment (beeme1mr) [inline readme.md:11]:
```suggestion
<h2 align="center">OpenFeature Python SDK</h2>
```

File: readme.md [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <picture>
+    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/white/openfeature-horizontal-white.svg">
+    <source media="(prefers-color-scheme: light)" srcset="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/black/openfeature-horizontal-black.svg">
+    <img align="center" alt="OpenFeature Logo">
+  </picture>
+</p>
+
+<!-- TODO: update title -->
```
Comment (beeme1mr) [inline readme.md:10]:
```suggestion
```

File: readme.md [hunk 4]
```diff
@@ -71,22 +72,112 @@ open_feature_api.set_provider(NoOpProvider())
...
+- Join us on [Slack](https://cloud-native.slack.com/archives/C0344AANLA1)
+- For more check out our [community page](https://openfeature.dev/community/)
+
+## 🤝 Contributing
+
+Interested in contributing? Great, we'd love your help! To get started, take a look at the [CONTRIBUTING](CONTRIBUTING.md) guide.
+
+### Thanks to everyone that has already contributed
+
+<!-- TODO: update with correct repo -->
```
Comment (beeme1mr) [inline readme.md:172]: Can this line be removed?

**Review by beeme1mr — APPROVED** [review]
Can you please add the license information to the bottom of the page. The rest looks great. Thanks for putting this together.

**Comment by jamescarr** [comment]
> Can you please add the license information to the bottom of the page. The rest looks great. Thanks for putting this together.

No problem, just updated to add the license and remove all the html comments for todos.

**Pull request merged**